openslides_backend/action/actions/mediafile/mixins.py

```diff
@@ -3,30 +3,25 @@
 from ....permissions.management_levels import OrganizationManagementLevel
 from ....permissions.permission_helper import has_organization_management_level
 from ....services.datastore.commands import GetManyRequest
-from ....shared.exceptions import (
-    ActionException,
-    MissingPermission,
-    PermissionException,
-)
+from ....shared.exceptions import ActionException, MissingPermission
 from ....shared.filters import And, Filter, FilterOperator, Not
 from ....shared.patterns import KEYSEPARATOR, Collection, FullQualifiedId
 from ...action import Action
 
 
-class MediafilePermissionMixin(Action):
+class MediafileMixin(Action):
     """
     Mixin to handle the check_permissions of mediafile actions.
-    Overwrite check_permissions() and get_meeting_id().
+    Overwrite update_instance(), check_permissions() and get_meeting_id().
     """
 
-    def check_permissions(self, instance: Dict[str, Any]) -> None:
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
         collection, id_ = self.get_owner_data(instance)
-
         self.check_parent_is_dir_and_owner(
             instance.get("parent_id"), str(instance.get("owner_id"))
         )
         parent_id = instance.get("parent_id")
-        if not parent_id and instance.get("id"):
+        if self.name == "mediafile.update":
```

Comment:
Mmh, I liked the more generic approach better (`if not parent_id`), so the mixin does not need to know which classes inherit from it.

Reply:
Yes, but the generic version doesn't work, as instance always has an id.

Reply:
Use a more generic approach for this issue. Replace the "mediafile.update".